flow/layers/container_layer.cc

```diff
@@ -246,10 +252,14 @@ ContainerLayer* MergedContainerLayer::GetChildContainer() const {
   return static_cast<ContainerLayer*>(layers()[0].get());
 }
 
-Layer* MergedContainerLayer::GetCacheableChild() const {
+CacheableLayer* MergedContainerLayer::GetCacheableChild() const {
   ContainerLayer* child_container = GetChildContainer();
   if (child_container->layers().size() == 1) {
-    return child_container->layers()[0].get();
+    auto* child = child_container->layers()[0].get();
+    if (child->IsCacheable()) {
+      return static_cast<CacheableLayer*>(child);
+    }
+    return nullptr;
```

Comment:
I don't think we should make this change, we shouldn't change its original logic.